### PR TITLE
I Reorganized deployment and test scripts, and fixed the delegate() bug

### PR DESCRIPTION
**Pull request by cartercameron1**
The auto deploy python script is now up to date. I moved the python file to the test folder. The only things you have to do before running auto deploy is npm install and starting your hardhat node. The scripts folder is cleaned; there is now one script for each contract. I also added the propose script back from one of the older versions for reference and allow additional testing for the governance contract. 

Note: currently setup-script.js only delegates the default address's ERC721 token.

### CODE REVIEW

**Comment by CptAstro** [comment]
Could you add comments to the hardcoded addresses in the scripts and tests files to show what address they're actually pointing to? I think I've got it down now, but it's not obvious what each address is especially since they change on redeployment

**Review by victor-wei126 — COMMENTED** [review]
The 3 deployment scripts for each contract look good, with some minor comments.

Regarding the 3 scripts added to `tests/`:
- `auto-deploy-script.py`: is this script in tests for the time being because it also runs the `setup-script`? I was thinking that it should be in the `scripts/` directory.
- `proposal-test-script.js`: why do we still need this script? I'm not sure what the purpose is since it's never run.
- `setup-script.js`: looks good, but maybe rename to make the purpose of it more clear. My understanding was that this script is to mint NFTs and delegate them, so frontend testers can use "members out of the box"

File: scripts/membershipNFT-deployment-script.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+async function main() {
+
+  const MembershipNFT = await hre.ethers.getContractFactory("MembershipNFT");
+  const membershipNFT = await MembershipNFT.deploy('UDAOToken', 'UDAO');
```
[the remark this answers is not in the document]
Comment (cartercameron1) [reply scripts/membershipNFT-deployment-script.js:4]: I actually open opened an issue regarding this yesterday. I was going to bring it up at the meeting. We can discuss official names of the tokens and contracts at the meeting.

File: test/setup-script.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+await mem.safeMint('0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC');
+await mem.safeMint('0x90F79bf6EB2c4f870365E785982E1f101E93b906');
+
+receiver = '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266'
+
+const receiverSigner = await ethers.getSigner(receiver);
+// call delegate with the receiver to set themselves as the vote delegate
+let tx2 = await mem.connect(receiverSigner).delegate(receiver);
+
+const Tok = await ethers.getContractFactory('UDAOToken');
```
[the remark this answers is not in the document]
Comment (cartercameron1) [reply test/setup-script.js:18]: It is for ease of testing after running the setup script it gives the governance contract erc20 tokens. Having a balance will let us test moving things around with more ease. I am going to edit the PR a bit adding comments about which address is which and some of the purposes behind various things.

**Comment by cartercameron1** [comment]
> Could you add comments to the hardcoded addresses in the scripts and tests files to show what address they're actually pointing to? I think I've got it down now, but it's not obvious what each address is especially since they change on redeployment

I will add additional comments on addresses and functionality. The addresses do not change on redeployment as long as you do them in the correct order. If you use the script it deploys them in the correct order.

File: test/setup-script.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (victor-wei126) [reply test/setup-script.js:18]: Great. There's no harm in transferring some erc20 tokens in the test script, so if you think it's necessary, we can keep it there.

[the remark this answers is not in the document]
Comment (cartercameron1) [reply test/setup-script.js:18]: It is not necessary but might speed up future testing.

**Pull request merged**